### ISSUE TITLE
[tool_crash] Invalid argument(s): Cannot find executable for /Users/xxxxxx/flutter/bin/cache/artifacts/engine/ios-release/gen_snapshot_x86_64.

### ISSUE THREAD

**Issue by bmyagmarjav** [flutter/flutter#54065]
## Command
```
flutter assemble --output=/Users/xxxxx/ss/myproject/ios/Flutter/ -dTargetPlatform=ios -dTargetFile=/Users/xxxxx/ss/myproject/main.beta.dart -dBuildMode=release -dIosArchs=x86_64 -dSplitDebugInfo= -dTreeShakeIcons=false -dTrackWidgetCreation=true -dDartObfuscation=false -dEnableBitcode= -dDartDefines= -dExtraFrontEndOptions= release_ios_bundle_flutter_assets
```

## Steps to Reproduce
1. ...
2. ...
3. ...

## Logs
ArgumentError: Invalid argument(s): Cannot find executable for /Users/xxxxx/flutter/bin/cache/artifacts/engine/ios-release/gen_snapshot_x86_64.
```
#0      _getExecutable (package:process/src/interface/local_process_manager.dart:127:5)
#1      LocalProcessManager.start (package:process/src/interface/local_process_manager.dart:43:30)
#2      _DefaultProcessUtils.start (package:flutter_tools/src/base/process.dart:462:28)
#3      _DefaultProcessUtils.stream (package:flutter_tools/src/base/process.dart:480:35)
#4      GenSnapshot.run (package:flutter_tools/src/base/build.dart:74:26)
#5      AOTSnapshotter.build (package:flutter_tools/src/base/build.dart:198:56)
#6      AotAssemblyBase.build (package:flutter_tools/src/build_system/targets/ios.dart:69:31)
#7      _BuildInstance._invokeInternal (package:flutter_tools/src/build_system/build_system.dart:614:25)
<asynchronous suspension>
#8      _BuildInstance.invokeTarget.<anonymous closure> (package:flutter_tools/src/build_system/build_system.dart:569:35)
#9      new Future.sync (dart:async/future.dart:224:31)
#10     AsyncMemoizer.runOnce (package:async/src/async_memoizer.dart:43:45)
#11     _BuildInstance.invokeTarget (package:flutter_tools/src/build_system/build_system.dart:569:21)
<asynchronous suspension>
<asynchronous suspension>
#12     BuildSystem.build (package:flutter_tools/src/build_system/build_system.dart:498:36)
#13     AssembleCommand.runCommand (package:flutter_tools/src/commands/assemble.dart:184:58)
#14     FlutterCommand.verifyThenRunCommand (package:flutter_tools/src/runner/flutter_command.dart:722:18)
<asynchronous suspension>
#15     FlutterCommand.run.<anonymous closure> (package:flutter_tools/src/runner/flutter_command.dart:619:33)
```
```
[✓] Flutter (Channel master, v1.17.1-pre.22, on Mac OS X 10.14.6 18G2022, locale en-US)
    • Flutter version 1.17.1-pre.22 at /Users/xxxxxx/flutter
    • Framework revision 14f3a36aba (2 days ago), 2020-04-03 17:06:01 -0700
    • Engine revision 09bc1fc45e
    • Dart version 2.8.0 (build 2.8.0-dev.20.0 e736495eb7)

[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.3)
    • Android SDK at /Users/xxxxx/Library/Android/sdk
    • Platform android-29, build-tools 29.0.3
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_212-release-1586-b4-5784211)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.3.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.3.1, Build version 11C504
    • CocoaPods version 1.8.4

[✓] Android Studio (version 3.6)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 44.0.2
    • Dart plugin version 192.7761
    • Java version OpenJDK Runtime Environment (build 1.8.0_212-release-1586-b4-5784211)

[✓] IntelliJ IDEA Community Edition (version 2019.2.4)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin version 44.0.2
    • Dart plugin version 192.7761

[✓] VS Code (version 1.43.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.9.0

[✓] Connected device (2 available)
    • Android SDK built for x86 • emulator-5554                        • android-x86 • Android 10 (API 29) (emulator)
    • iPhone 11 Pro Max         • 51ADEEE6-00B3-48ED-B6A0-366359CF3CB6 • ios         • com.apple.CoreSimulator.SimRuntime.iOS-13-3 (simulator)

• No issues found!

```

## Flutter Application Metadata
**Type**: app
**Version**: 0.0.1+1
**Material**: true
**Android X**: false
**Module**: false
**Plugin**: false
**Android package**: null
**iOS bundle identifier**: null
**Creation channel**: master
**Creation framework version**: ba0df45c12f008e9ef3a8d90cdb15bc91e2f4221
### Plugins
cloud_firestore-0.13.4+2
cloud_firestore_web-0.1.1+2
firebase_analytics-5.0.11
firebase_auth-0.15.5+3
firebase_auth_web-0.1.2
firebase_core-0.4.4+3
firebase_core_web-0.1.1+2
firebase_messaging-6.0.13
firebase_remote_config-0.3.0+3
firebase_storage-3.1.5
flutter_local_notifications-1.3.0
shared_preferences-0.5.6+3
shared_preferences_macos-0.0.1+6
shared_preferences_web-0.1.2+4
url_launcher-5.4.2
url_launcher_macos-0.0.1+4
url_launcher_web-0.1.1+1

**Comment by edSarti** [comment]
The almost same issue happened to me yesterday (2020 May, 21), after receiving a message on the "terminal / Android Studio" to perform a "flutter upgrade". After executing the command, the Flutter crashes completely. Incredibly, not even the "flutter doctor" worked anymore, which for me is kind of unacceptable. 

I didn't know that the "flutter upgrade" did not consider the channel being worked on. Before the crash, I had the beta channel version (for Web dev) and the command updated to the latest version of the "stable channel". The link with Git has been broken.

I tried to find something like a "flutter unupgrade" to revert the update, but it seems that it doesn't exist (yet). I think it is important that an update can be reversed in these cases.

After literally hours of searching for this issue, my drastic solution was to REMOVE and REINSTALL THE FLUTTER COMPLETELY, from the beginning and redo the "flutter upgrade" on the beta channel. Operationally speaking, the problem was solved, but I think that Flutter people should "check the channel in use by the user, before even suggesting a" flutter upgrade ", and in doing so, do it on the right channel", at least until solve the problem really.

And, also, it is important that the "flutter doctor" is a completely segregated process and that it does not crash together with the Flutter. As it is, it doesn't make sense to me.

Trying to contribute, thank you!
